homeassistant/components/group/__init__.py

```diff
@@ -1,11 +1,22 @@
 """Provide the functionality to group entities."""
 import asyncio
+from itertools import chain
 import logging
 from typing import Any, Iterable, List, Optional, cast
 
 import voluptuous as vol
 
 from homeassistant import core as ha
+from homeassistant.components.climate.const import HVAC_MODE_OFF, HVAC_MODES
+from homeassistant.components.vacuum import STATE_CLEANING, STATE_ERROR, STATE_RETURNING
+from homeassistant.components.water_heater import (
```

Comment:
I am not sure if it makes sense for the group integration to know about every other platform, or that we should use [the integration platform helper](https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/integration_platform.py) to allow integrations to teach the group integration about grouping.

Reply:
Integration helper makes the most sense to make this scale in the future as I'm sure we will want to add more of these later

Reply:
I think I need to update `after_dependencies` as well

Reply:
```
2020-09-28 20:54:18 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: entity_id=group.outside_non_security_lights_and_switches, old_state=None, new_state=<state group.outside_non_security_lights_and_switches=unknown; entity_id=('light.outside_living_room_overhead', 'light.outside_east_lot_lights', 'light.outside_patio_lights_g', 'light.outside_pool_wall_lights', 'switch.outside_patio_fans'), order=0, friendly_name=Outside Non Security Lights and Switches @ 2020-09-28T15:54:18.700359-05:00>>
2020-09-28 20:54:20 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: entity_id=group.outside_non_security_lights_and_switches, old_state=<state group.outside_non_security_lights_and_switches=unknown; entity_id=('light.outside_living_room_overhead', 'light.outside_east_lot_lights', 'light.outside_patio_lights_g', 'light.outside_pool_wall_lights', 'switch.outside_patio_fans'), order=0, friendly_name=Outside Non Security Lights and Switches @ 2020-09-28T15:54:18.700359-05:00>, new_state=<state group.outside_non_security_lights_and_switches=off; entity_id=('light.outside_living_room_overhead', 'light.outside_east_lot_lights', 'light.outside_patio_lights_g', 'light.outside_pool_wall_lights', 'switch.outside_patio_fans'), order=0, friendly_name=Outside Non Security Lights and Switches @ 2020-09-28T15:54:20.618864-05:00>>
2020-09-28 20:54:30 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: entity_id=group.outside_non_security_lights_and_switches, old_state=<state group.outside_non_security_lights_and_switches=off; entity_id=('light.outside_living_room_overhead', 'light.outside_east_lot_lights', 'light.outside_patio_lights_g', 'light.outside_pool_wall_lights', 'switch.outside_patio_fans'), order=0, friendly_name=Outside Non Security Lights and Switches @ 2020-09-28T15:54:20.618864-05:00>, new_state=<state group.outside_non_security_lights_and_switches=on; entity_id=('light.outside_living_room_overhead', 'light.outside_east_lot_lights', 'light.outside_patio_lights_g', 'light.outside_pool_wall_lights', 'switch.outside_patio_fans'), order=0, friendly_name=Outside Non Security Lights and Switches @ 2020-09-28T15:54:30.753434-05:00>>
```

First we get `STATE_UNKNOWN`, then the `light` integration loads and we get `STATE_OFF`, then the integration providing the lights load and we get `STATE_ON`.

I wonder if this should defer tracking states until the start event like we do with `template`

Reply:
Since the states weren't loaded into the state machine in the same order each time sometimes the group groups from
STATE_UNKNOWN to STATE_OFF to STATE_ON, and sometimes it goes from STATE_UNKOWN to STATE_ON.  I've tweaked it so it doesn't activate tracking until the start event so we don't end up with unexpected states.